kroxylicious-kms-provider-hashicorp-vault-test-support/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/AbstractVaultTestKmsFacade.java

```diff
@@ -155,6 +157,7 @@ public void rotateKek(String alias) {
 
             if (exists(alias)) {
                 rotate(alias);
+                expectedVersion++;
```

Comment:
No, this is a bad approach.  You are assuming an order in which the caller invokes the API methods.  But the caller is free to call the API methods in any order they please.  Unfortunately the `assertKekVersionWithinParcel` logic has to go in the RecordEncryptionST.  It knows the state of the Manager (because it is in charger), so it has enough knowledge to be able to make the assertion.